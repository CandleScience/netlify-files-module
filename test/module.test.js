jest.setTimeout(60000)

const { resolve, join } = require('path')
const { readFileSync } = require('fs')
const { Nuxt, Builder, Generator } = require('nuxt-edge')
const config = require('../example/nuxt.config')
config.dev = false

let nuxt

describe('basic', () => {
  beforeAll(async () => {
    nuxt = new Nuxt(config)
    await nuxt.ready()
    const builder = await new Builder(nuxt)
    const generator = new Generator(nuxt, builder)
    await generator.generate()
  })

  afterAll(async () => {
    await nuxt.close()
  })

  test('it copies netlify files file', () => {
    const distBasePath = resolve(nuxt.options.rootDir, join(nuxt.options.generate.dir))
    const headersPath = join(distBasePath, '_headers')
    const redirectsPath = join(distBasePath, '_redirects')

    expect(readFileSync(headersPath, { encoding: 'utf8' })).toMatchSnapshot()
    expect(readFileSync(redirectsPath, { encoding: 'utf8' })).toMatchSnapshot()
  })
})
