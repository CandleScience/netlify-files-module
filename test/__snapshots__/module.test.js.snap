// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic it copies netlify files file 1`] = `
"foobar
"
`;

exports[`basic it copies netlify files file 2`] = `
"barfoo
"
`;
